src/network.rs

```diff
@@ -610,15 +626,36 @@ impl<'de> Deserialize<'de> for ActivePeer {
     }
 }
 
-fn derive_key_pair(seed: &Seed) -> libp2p::identity::Keypair {
-    let mut engine = sha256::HashEngine::default();
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Seed([u8; SEED_LENGTH]);
+
+impl Seed {
+    /// prefix "NETWORK" to the provided seed and apply sha256
+    pub fn new(seed: [u8; crate::seed::SEED_LENGTH]) -> Self {
+        let mut engine = sha256::HashEngine::default();
+
+        engine.input(&seed);
+        engine.input(b"NETWORK");
 
-    engine.input(&seed.bytes());
-    engine.input(b"LIBP2P_KEYPAIR");
+        let hash = sha256::Hash::from_engine(engine);
+        Self(hash.into_inner())
+    }
 
-    let hash = sha256::Hash::from_engine(engine);
-    let key = ed25519::SecretKey::from_bytes(hash.into_inner()).expect("we always pass 32 bytes");
-    libp2p::identity::Keypair::Ed25519(key.into())
+    pub fn bytes(&self) -> [u8; SEED_LENGTH] {
+        self.0
+    }
+
+    pub fn derive_key_pair(&self) -> libp2p::identity::Keypair {
```

Comment:
Call this `derive_libp2p_identity` maybe?

Reply:
hehe, yes this was the name in one of my patches. 👍